### PR TITLE
implement f32

### PR DESCRIPTION
**Pull request by woojinnn**
## Work
Implementation of issue [#623](https://github.com/gluesql/gluesql/issues/623)

## Need your help:  
There are some loss of information while casting, which is expected behavior in Rust.  
Should we endure this issue?  
E.g., value `1234567890` is casted to `1234568000`.

### CODE REVIEW

File: core/src/data/value/json.rs [hunk 1]
```diff
@@ -43,6 +43,7 @@ impl TryFrom<Value> for JsonValue {
             Value::I128(v) => JsonNumber::from_str(&v.to_string())
                 .map(JsonValue::Number)
                 .map_err(|_| ValueError::UnreachableJsonNumberParseFailure(v.to_string()).into()),
+            Value::F32(v) => Ok(JsonValue::Number(JsonNumber::from_f64(v as f64).unwrap())),
```
Comment (panarch) [inline core/src/data/value/json.rs:46]: 1. We can use this
https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#impl-From%3Cf32%3E-for-Value

`Ok(v.into())` would be enough

2. unit test for `Value::F32` to `JsonValue::Number`

File: core/src/data/key.rs [hunk 2]
```diff
@@ -84,6 +84,7 @@ impl TryFrom<Value> for Key {
             Uuid(v) => Ok(Key::Uuid(v)),
             Decimal(v) => Ok(Key::Decimal(v)),
             Null => Ok(Key::None),
+            F32(_) => Err(KeyError::FloatTypeKeyNotSupported.into()),
             F64(_) => Err(KeyError::FloatTypeKeyNotSupported.into()),
```
Comment (panarch) [inline core/src/data/key.rs:88]: Same error so we can merge this into same branch
`F32(_) | F64(_) => Err(KeyError::FloatTypeKeyNotSupported.into()),`

File: core/src/data/value/mod.rs [hunk 3]
```diff
@@ -116,6 +120,7 @@ impl Value {
             Value::I32(_) => Some(DataType::Int32),
             Value::I64(_) => Some(DataType::Int),
             Value::I128(_) => Some(DataType::Int128),
+            Value::F32(_) => Some(DataType::Float32),
```
Comment (panarch) [inline core/src/data/value/mod.rs:123]: There's an unit test with `get_type`, we can add `F32` test to that.

**Review by panarch — CHANGES_REQUESTED** [review]
Thanks a lot, let's add some tests for covering newly added lines.

Coveralls report would be quite useful.
You can click the link and that will show uncovered line with code changes

image

e.g.
image

**Comment by pythonbrad** [comment]
@woojinnn, please Is the implementation complete?

**Comment by pythonbrad** [comment]
@woojinnn, please Is the implementation complete?

**Comment by woojinnn** [comment]
@pythonbrad Nah. Feel free to take this issue if you're interested

**Comment by ever0de** [comment]
closed by #1145

**Pull request closed without merging**